wgpu/src/lib.rs

```diff
@@ -3447,13 +3447,32 @@ impl Drop for SurfaceTexture {
 
 impl Surface {
     /// Returns a vec of supported texture formats to use for the [`Surface`] with this adapter.
-    /// Note: The first format in the vector is preferred
     ///
     /// Returns None if the surface is incompatible with the adapter.
     pub fn get_supported_formats(&self, adapter: &Adapter) -> Option<Vec<TextureFormat>> {
         Context::surface_get_supported_formats(&*self.context, &self.id, &adapter.id)
     }
 
+    /// Returns an optimal texture format to use for the [`Surface`] with this adapter.
+    pub fn get_preferred_format(&self, adapter: &Adapter) -> Option<wgt::TextureFormat> {
```

Comment:
Instead of doing this separately, both methods of "preferred" format should return the same value. How about inside of get_supported_formats we just bring the srgb formats to the front - `caps.format.sort_by_key(|f| !f.describe().srgb `. Once we support texture view changing srgb-ness we can remove the sort.